### ISSUE TITLE
Got some errors while load the script for the first time.

### ISSUE THREAD

**Issue by Deepnerd** [DanWin/le-chat-php#65]
****[Screenshot]**** https://imgur.com/a/JDWUkLl

Warning: Use of undefined constant SODIUM_CRYPTO_AEAD_AES256GCM_KEYBYTES - assumed 'SODIUM_CRYPTO_AEAD_AES256GCM_KEYBYTES' (this will throw an Error in a future version of PHP) in C:\xampp\htdocs\index.php on line 4233

Warning: Use of undefined constant SODIUM_CRYPTO_AEAD_AES256GCM_KEYBYTES - assumed 'SODIUM_CRYPTO_AEAD_AES256GCM_KEYBYTES' (this will throw an Error in a future version of PHP) in C:\xampp\htdocs\index.php on line 4234

Warning: substr() expects parameter 3 to be int, string given in C:\xampp\htdocs\index.php on line 4234

Warning: Use of undefined constant SODIUM_CRYPTO_AEAD_AES256GCM_NPUBBYTES - assumed 'SODIUM_CRYPTO_AEAD_AES256GCM_NPUBBYTES' (this will throw an Error in a future version of PHP) in C:\xampp\htdocs\index.php on line 4238

Warning: Use of undefined constant SODIUM_CRYPTO_AEAD_AES256GCM_NPUBBYTES - assumed 'SODIUM_CRYPTO_AEAD_AES256GCM_NPUBBYTES' (this will throw an Error in a future version of PHP) in C:\xampp\htdocs\index.php on line 4239

Warning: substr() expects parameter 3 to be int, string given in C:\xampp\htdocs\index.php on line 4239

****[The Old version of this chat script is working without any errors.]****
****[Renamed the chat.php to index.php]****
*****[My configurations for the new updated script]*****

function load_config(){
	mb_internal_encoding('UTF-8');
	define('VERSION', '1.24'); // Script version
	define('DBVERSION', 42); // Database layout version
	define('MSGENCRYPTED', true); // Store messages encrypted in the database to prevent other database users from reading them - true/false - visit the setup page after editing!
	define('ENCRYPTKEY_PASS', '13117132423542543653635467854326'); // Recommended length: 32. Encryption key for messages
    define('AES_IV_PASS', '012345678912'); // Recommended length: 12. AES Encryption IV
	define('DBHOST', 'localhost'); // Database host
	define('DBUSER', 'root'); // Database user
	define('DBPASS', ''); // Database password
	define('DBNAME', 'newupdate'); // Database
	define('PERSISTENT', true); // Use persistent database conection true/false
	define('PREFIX', ''); // Prefix - Set this to a unique value for every chat, if you have more than 1 chats on the same database or domain - use only alpha-numeric values (A-Z, a-z, 0-9, or _) other symbols might break the queries
	define('MEMCACHED', false); // Enable/disable memcached caching true/false - needs memcached extension and a memcached server.

**Comment by DanWin** [comment]
Most likely you are using an outdated version of PHP. Please upgrade to at least PHP7.2, preferably PHP7.4. See https://www.php.net/supported-versions.php for currently maintained versions.
But good point, a user with deprecated PHP versions should not see a bunch of warnings he may not understand. I will add a version check to make sure people upgrade PHP.

**Comment by Deepnerd** [comment]
Thanks daniel! i miss your DansChat and your  Hosting :/

**Comment by Deepnerd** [comment]
Hey dani, did you fixed that white flash screen when page refresh every time issue? i got that issue in your early version of script. and now im in your chat.danwin1210.me and page refresh as usual an that white flash issue gone. fixed? or my local webserver problem?

**Comment by DanWin** [comment]
This probably related to buffering. If your web browser is displaying content faster than it is receiving (because the server is too slow), it will at given time not yet have applied the CSS style, which may not yet have fully loaded. You can turn on output buffering, which will block on the server, until all content is generated. Once it is ready, it will be sent to the client in full, which should prevent the flashing screen. If you have an external CSS file included, this may also be related to that, if it has to be re-validated due to caching policies.

**Comment by cypherbits** [comment]
@Deepnerd my fault here since I spent some days ago hardening the security of this chat in many ways. This involved upgrading to a newer, safer and faster message encryption. But to use this new encryption you should install at least PHP version 7.2 or better.

**Comment by Deepnerd** [comment]
Still im getting the same error ****php version 7.4.5 (XAMPP)****

**Comment by cypherbits** [comment]
> Still im getting the same error ****php version 7.4.5 (Latest XAMPP)****
> i have sodium extension in my php.ini ****(;extension=sodium)****
> 

A dumb question... Did you enabled it by removing the ';' ??

And then restarting xampp?

Try this too:

`In \xampp\apache\conf\extra\httpd-xampp.conf

Find:
LoadFile "/xampp/php/php7ts.dll"
LoadFile "/xampp/php/libpq.dll"
LoadModule php7_module "/xampp/php/php7apache2_4.dll"


Add:
LoadFile "/xampp/php/libsodium.dll"

`

**Comment by Deepnerd** [comment]
Finally it worked! But no offense normal  users doesn't know is it enabled or disabled. so please add some instructions how to fix this error if user got this error. because of of them testing this first in their own localhost and change styles and then to the server. thanks for your help. appreciate that!

**Comment by Deepnerd** [comment]
Issue fixed! closed.

**Comment by Deepnerd** [comment]
Got another issue: when i submit my first message to the chat 
Error ****Apache HTTP server has stopped working****
Screenshot: https://imgur.com/a/5MBQuqX

**Comment by Deepnerd** [comment]
Got another issue: when i submit my first message to the chat
Error Apache HTTP server has stopped working
Screenshot: https://imgur.com/a/5MBQuqX

Im going to try this on ubuntu webserver. i think there are many issues to run this in windows.
But old version works without any problem.

And also do you have any recommended software to use in windows? without xampp? any other?

**Comment by cypherbits** [comment]
> Finally it worked! But no offense normal  users doesn't know is it enabled or disabled or how to fix this. so please add some instructions how to fix this error if user got this error. because most of them testing this script first in their own localhost and change styles and then to the server. thanks for your help. appreciate that!  

Well, official PHP documentation says sodium is included and enabled by default since PHP 7.2 
It seems it is an issue with Xampp...

**Comment by cypherbits** [comment]
> Got another issue: when i submit my first message to the chat
> Error Apache HTTP server has stopped working
> Screenshot: https://imgur.com/a/5MBQuqX
> 
> Im going to try this on ubuntu webserver. i think there are many issues to run this in windows.
> But old version works without any problem.
> My xampp works fine on other scripts.
> 
> And also do you have any recommended software to use in windows? without xampp? any other?
> This error will get windows vps users. :/

No idea on this one... This is one of the reasons I don't use Windows. Xampp gave me many headaches where php on linux works fine by default...

**Comment by DanWin** [comment]
> Got another issue: when i submit my first message to the chat
> Error Apache HTTP server has stopped working
> Screenshot: https://imgur.com/a/5MBQuqX
> 
> Im going to try this on ubuntu webserver. i think there are many issues to run this in windows.
> But old version works without any problem.
> My xampp works fine on other scripts.

Perhaps the issue is with the sodium extension for XAMPP, you just enabled. Other scripts weren't using the features it provides, so there were no issues. The problem itself is however unrelated to the chat script, as it only triggers the crash. But the cause of it lies somewhere in XAMPP.

> And also do you have any recommended software to use in windows? without xampp? any other?
> This error will get windows vps users. :/

Another alternative could be https://www.mamp.info/en/mamp/windows/

**Comment by Deepnerd** [comment]
Thank you guys! sorry for being trouble for you guys. thank you... :)

**Comment by Deepnerd** [comment]
@DanWin Hey dani, can you please explain to me how to get removed white flash screen problem when page reloads every time. im using vps :-

CPU: 1 vCore
RAM: 1024 MB
Ubuntu 20.04 x64

only to do enable output buffering in php.ini?
